pkg/operator/target_config_reconciler.go

```diff
@@ -117,35 +115,7 @@ func (c TargetConfigReconciler) sync() error {
 		return fmt.Errorf("descheduler should have an interval set")
 	}
 
-	forceDeployment := false
-	_, forceDeployment, err = c.manageConfigMap(descheduler)
```

Comment:
This needs to be handled separately to `ApplyDeployment` as a new annotation. Since the hash calculated from the deployment itself does not include changes in the external dependent resources. There are two cases to take into account:
1. config map changes: the deployment needs to be annotated with `"configmaps/cluster": configMap.ObjectMeta.ResourceVersion` to trigger additional update of the deployment
2. if err != nil, the deployment needs to be scaled down: the original code needs to be kept as it is. Just, without passing `forceDeployment` into `manageDeployment`. `forceDeployment` will be used only for scaling the deployment to `0`.

Reply:
I am annotating all the other dependent resources (services, roles, sa, etc). Let me know if we just want to annotate for the configMap

Reply:
+1 for annotating the other resources